types/address.go

```diff
@@ -111,7 +111,7 @@ func VerifyAddressFormat(bz []byte) error {
 		return verifier(bz)
 	}
 	if len(bz) != AddrLen {
-		return errors.New("incorrect address length")
+		return fmt.Errorf("incorrect address length (expected: %d, actual: %d)", AddrLen, len(bz))
```

Comment:
Actually, this should ideally be a `types/errors` / `sdkerrors` object, so it has a proper ABCI code and can be passed up as a base error message.

@marbar3778

Reply:
Wouldn't this require to change the type of all the `errors.New`s in this file (which I assume are from https://golang.org/pkg/errors/#New) as well as using a stricter return interface?

Reply:
They are all `error` interfaces as well. They just carry ABCI info along with them, so they display better (don't become internal error)

Reply:
But yeah, this is everywhere in this file already and should be another issue.